include/alpaka/ctx/block/CtxBlockOacc.hpp

```diff
@@ -113,8 +113,8 @@ namespace alpaka
                 }
 #    pragma acc atomic capture
                 {
-                    ++m_syncCounter[slot];
-                    sum = m_syncCounter[slot];
+                    ++m_syncCounter[slot + 1];
+                    sum = m_syncCounter[slot + 1];
```

Comment:
IMO there is a data race between this line and line 106. @jkelling I contacted you over Mattermost for a VC to peer review it on a shared screen.

Reply:
Ahh line 106 is setting `nextSlot` and not `slot`, then it should be ok.

Reply:
Yes, `nextSlot` is for the next barrier. The loop in line 119 is protecting that .. or not without this fix.